thingif/src/main/java/com/kii/thingif/temporal/base/BaseAnd.java

```diff
@@ -0,0 +1,8 @@
+package com.kii.thingif.temporal.base;
+
+import java.util.List;
+
+public interface BaseAnd<T> {
```

Comment:
> I expect AndClauseInQuery and AndClauseInTrigger solve the problem

Can you show how those sub class solve the problem? 

> developer should not directly use BaseAnd

I do not assume so neither...  And i think i don't show that BaseAnd is directly used in my sketch.